lib/Doctrine/Collection.php

```diff
@@ -571,7 +571,11 @@ public function loadRelated($name = null)
             return null;
         }
 
-        $dql     = $rel->getRelationDql(count($list), 'collection');
+        if ($rel instanceof Doctrine_Relation_Association) {
```

Comment:
branching here to only use the second parameter if the $rel var is of a certain object type, as the `getRelationDql` method signature is different depending on the relation object.